Virgil/1 two-sum.cpp

```diff
@@ -0,0 +1,18 @@
+#include <vector>
+#include <iostream>
+#include <algorithm>
+using namespace std;
+class Solution {
+public:
+    vector<int> twoSum(vector<int>& nums, int target) {
+        int n = nums.size();
+        for (int i = 0; i < n-1; i++) {
+            for (int j = i + 1; j < n; j++) {
+                if (nums[i] + nums[j] == target) {
+                    return {i, j};
+                }
+            }
+        }
+        return {};

```

Comment:
This is a time O(n<sup>2</sup>) space O(1) program.

Can you rewrite one with O(n) space and time?